qir/qat/PreTransformValidation/PreTransformValidationPass.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "PreTransformValidation/PreTransformValidationPass.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    PreTransformValidationPass::PreTransformValidationPass(
+        PreTransformValidationPassConfiguration const& cfg,
+        ILoggerPtr const&                              logger)
+      : config_{cfg}
+      , logger_{logger}
+    {
+    }
+
+    llvm::PreservedAnalyses PreTransformValidationPass::run(llvm::Module& module, llvm::ModuleAnalysisManager& /*mam*/)
+    {
+        uint64_t function_count{0};
+
+        if (config_.requireStraightLineCode())
+        {
+            for (auto& function : module)
```

Comment:
Consider moving `function_count` to inside of `if`. Otherwise, if `if` is not entered then the allocation and initialization in line 26 is redundant.
```suggestion
        if (config_.requireStraightLineCode())
        {
            uint64_t function_count{0};

            for (auto& function : module)
```